Lib/fontTools/otlLib/builder.py

```diff
@@ -2509,9 +2509,14 @@ def buildAttachPoint(points):
 
 def buildCaretValueForCoord(coord):
     # 500 --> otTables.CaretValue, format 1
+    # (500, DeviceTable) --> otTables.CaretValue, format 3
     self = ot.CaretValue()
-    self.Format = 1
-    self.Coordinate = coord
+    if isinstance(coord, tuple):
```

Comment:
instead of optionally taking a 2-tuple, how about adding an additional deviceTable=None parameter, when not None you'd make Format 3 table?

Reply:
This would only move the optional tuple down to `buildLigGlyph` and make the code their slightly uglier without changing much. `buildLigGlyph` takes a list of coordinates, so if were were to split off the devices we would need to lists and keep them in sync (e.g. when we sort the coordinates).